modules/features/manatime_feature_timesheet/manatime_feature_timesheet.module

```diff
@@ -0,0 +1,46 @@
+<?php
+/**
+ * @file
+ * Code for the Manatime timesheet feature.
+ */
+
+include_once 'manatime_feature_timesheet.features.inc';
+
+/**
+ * Implements hook_cron().
+ */
+function manatime_feature_timesheet_cron() {
+
+  $time_execution = variable_get('manatime_feature_timesheet_execution_time', 0);
+
+  if (REQUEST_TIME >= $time_execution) {
+
+    $next_execution_time = $time_execution + 604800;
+    $begining_date = format_date(REQUEST_TIME, 'custom', 'F d, Y');
```

Comment:
I think REQUEST_TIME is not the right beginning date. Request time it's supposed to be sunday, but your beginning date is supposed to be monday.
Btw, typo in variable name.